example-proxy.ini

```diff
@@ -65,4 +63,18 @@ attribute = sAMAccountName
 # The timeout is specified in seconds.
 # This feature is EXPERIMENTAL.
 enabled = false
-timeout = 3
\ No newline at end of file
+timeout = 3
+
+[preamble-cache]
```

Comment:
Well, the implementation is structured as follows:
If the preamble cache is enabled, the LDAP proxy monitors incoming search requests and their responses from the LDAP backend. If the request's filter contains an "app marker" (e.g. ``foo`` in ``(objectclass=App-foo)`` and the response contains exactly one entry, the tuple ``(dn, marker)`` is added to the preamble cache, from which it gets evicted automatically after a timeout.

Let's assume the user now enables the ``preamble`` realm mapping strategy. For an incoming bind request with a DN, the strategy queries the preamble cache. If there is a matching entry, the strategy retrieves the cached app marker and maps it to a realm (according to the ``[[mapping]]`` config section).

So, the preamble cache itself knows nothing about realms, it only stores DNs and app markers. Technically, the user could enable the preamble cache without using the ``preamble`` realm mapping strategy (althought the preamble cache would not be very useful in this case).

I propose the following: Internally, we keep the name "preamble" to describe the LDAP search prior to a bind request. We rename to "preamble cache" to "app cache" (which is a nice term, because it caches app markers). So the user never has to deal with the term "preamble".